packages/components/pie-button/test/helpers/page-object/pie-button-form-integration.page.ts

```diff
@@ -0,0 +1,188 @@
+        this.userPaymentCardTypeSelect = this.formLocator.getByTestId(form.selectors.userPaymentCardType.dataTestId);
+        this.userPaymentCardNumberInput = this.formLocator.getByTestId(form.selectors.userPaymentCardNumber.dataTestId);
+        this.userPaymentCardExpirationInput = this.formLocator.getByTestId(form.selectors.userPaymentCardExpiration.dataTestId);
+
+        this.resetPieButton = this.formLocator.locator(`[data-test-id="${form.selectors.pieButtonReset.dataTestId}"]`);
+        this.resetNativeButton = this.formLocator.locator(`[data-test-id="${form.selectors.nativeButtonReset.dataTestId}"]`);
+        this.submitPieButton = this.formLocator.locator(`[data-test-id="${form.selectors.pieButtonSubmit.dataTestId}"]`);
+
+        this.formSubmittedFlag = this.page.getByTestId(form.selectors.formSubmittedFlag.dataTestId);
+    }
+
+    /**
+     * Fills out the form with the provided input data.
+     *
+     * @param {FormInput} inputData - The data to fill the form with.
+     * @returns {Promise<FormIntegrationPage>} A Promise that resolves with the current instance once the form has been filled.
+     */
+    async fillForm (inputData: FormInput): Promise<FormIntegrationPage> {
+        await this.userNameInput.fill(inputData.userName);
+        await this.userEmailInput.fill(inputData.userEmail);
+        await this.userPasswordInput.fill(inputData.userPassword);
+        await this.userPaymentCardTypeSelect.selectOption(inputData.userPaymentCardType);
+        await this.userPaymentCardNumberInput.fill(inputData.userPaymentCardNumber);
+        await this.userPaymentCardExpirationInput.fill(inputData.userPaymentCardExpiration);
+
+        return this;
+    }
+
+    /**
+     * Clicks the specified form button.
+     *
+     * @param {FormButtonType} buttonType - The type of the button to click.
+     * @returns {Promise<void>} A Promise that resolves once the button has been clicked.
+     */
+    async clickFormButton (buttonType: FormButtonType): Promise<void> {
+        switch (buttonType) {
+            case 'pie-button-submit':
+                await this.submitPieButton.click();
+                break;
+            case 'pie-button-reset':
+                await this.resetPieButton.click();
+                break;
+            case 'native-reset':
+                await this.resetNativeButton.click();
+                break;
+            default:
+                throw new Error(`Invalid button type: ${buttonType}`);
+        }
+    }
+
+    /**
+     * Focuses on the specified field.
+     *
+     * @param {keyof FormInput} fieldName - The name of the field to focus on.
+     * @returns {Promise<Page>} A Promise that resolves once the field has been focused.
+     */
+    async focusField (fieldName: keyof FormInput): Promise<Locator> {
+        switch (fieldName) {
+            case 'userName':
+                await this.userNameInput.focus();
+                return this.userNameInput;
```

Comment:
Does this need to return the field?
And if not, do we need this `focusField` method at all or could we just make the selectors public and then people can just call `focus()` directly on those?

Reply:
Yeah I opted to remove this. It does go again POM best practice making the selectors public, but It definitely makes things easier by being able to interact with the native Playwright API in the tests.